docs/src/logging.md

```diff
@@ -0,0 +1,159 @@
+# Logging
+
+## Overview
+
+Logging is a crucial tool for debugging, monitoring, and understanding the behavior of your applications. ClimaComms extends Julia's built-in logging functionality (provided by [`Logging.jl`](https://docs.julialang.org/en/v1/stdlib/Logging/)) to work seamlessly in distributed computing environments, particularly with MPI.
+
+### Julia's Logging System
+
+Julia's standard library provides a flexible logging system through `Logging.jl`. Key concepts include:
+
+- **Logger**: An object that handles log messages, determining how they are formatted and where they are sent
+- **Log Level**: Indicates the severity/importance of a message
+- **Log Record**: Contains the message and associated metadata (level, module, line number, etc.)
+
+The default logger in Julia (as of v1.11) is `Logging.ConsoleLogger()`, which prints messages to the console. You can check your current logger using:
+
+```julia
+using Logging; current_logger()
+```
+
+## ClimaComms Logging Features
+
+ClimaComms builds upon Julia's logging system by providing specialized loggers for distributed computing:
+
+### OnlyRootLogger
+
+`OnlyRootLogger(context)` returns a logger that silences non-root processes.
+If using MPI, this logger is enabled on the first `init` call.
+
+### FileLogger
+
+`FileLogger(context, log_dir)` writes to `stdout` and to a file `log_dir/output.log` simultaneously.
+If using MPI, the `FileLogger` separates logs by MPI process into different files, so that process `i` will write to the `rank_$i.log` file in the `log_dir` directory. In this case, `log_dir/output.log` is a [symbolic link](https://en.wikipedia.org/wiki/Symbolic_link) to the root process logs.
```

Comment:
```suggestion
If using MPI, the `FileLogger` separates logs by MPI process into different files, so that process `i` will write to the `rank_$i.log` file in the `$log_dir` directory, where `$log_dir` is of your choosing. In this case, `$log_dir/output.log` is a [symbolic link](https://en.wikipedia.org/wiki/Symbolic_link) to the root process logs. In other words, you can always look at `$log_dir/output.log` for the output. 
```